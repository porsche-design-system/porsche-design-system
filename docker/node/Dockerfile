FROM mcr.microsoft.com/playwright:v1.26.0-focal

ENV PATH="/opt/porsche-design-system/bin:${PATH}"

RUN apt-get update && \
    apt-get install -yq gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 \
    libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 \
    libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 \
    libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 \
    ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils libnotify-dev \
    libgtk2.0-0 xvfb rsync openssh-client procps git curl unzip && \
    rm -rf /var/lib/apt/lists/*

# https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-linux.html#cliv2-linux-install
RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" && \
    unzip awscliv2.zip && \
    ./aws/install && \
    rm -rf awscliv2.zip ./aws/install

COPY ./entrypoint.sh "/opt/porsche-design-system/entrypoint.sh"
COPY ./bin "/opt/porsche-design-system/bin"
COPY ./shared "/opt/porsche-design-system/shared"

ENTRYPOINT [ "/opt/porsche-design-system/entrypoint.sh" ]
