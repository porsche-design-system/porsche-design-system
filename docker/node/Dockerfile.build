FROM docker.pkg.github.com/porscheui/porsche-design-system/node:12.16.1-stretch

# COPY package.json yarn.lock .npmrc /opt/porsche-design-system-app/
# COPY packages /opt/porsche-design-system-app/packages/

# WORKDIR /opt/porsche-design-system-app

ARG PORSCHE_NPM_REGISTRY_TOKEN
ENV PORSCHE_NPM_REGISTRY_TOKEN=$PORSCHE_NPM_REGISTRY_TOKEN

RUN exec "run-install"
