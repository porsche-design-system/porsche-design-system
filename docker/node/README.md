# Docker Image

### Fonts

Make sure fonts listed in directory `./fonts` are only stored in private repository and/or container registry for internal usage because of licensing.