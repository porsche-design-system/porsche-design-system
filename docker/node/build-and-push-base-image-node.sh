#!/bin/bash

REGISTRY=ghcr.io/porscheui/porsche-design-system
IMAGE=node
TAG=16.15.1-stretch-slim

docker pull $REGISTRY/$IMAGE:latest

docker build -t $REGISTRY/$IMAGE:$TAG -t $REGISTRY/$IMAGE:latest .

docker push $REGISTRY/$IMAGE:$TAG
docker push $REGISTRY/$IMAGE:latest
