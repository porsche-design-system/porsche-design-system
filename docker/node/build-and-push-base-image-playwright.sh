#!/bin/bash

REGISTRY=ghcr.io/porsche-design-system/porsche-design-system
IMAGE=playwright
TAG=v1.27.0-focal-v3-alpha-6

docker pull $REGISTRY/$IMAGE:latest

docker build -f Dockerfile-playwright -t $REGISTRY/$IMAGE:$TAG -t $REGISTRY/$IMAGE:latest .

docker push $REGISTRY/$IMAGE:$TAG
docker push $REGISTRY/$IMAGE:latest
