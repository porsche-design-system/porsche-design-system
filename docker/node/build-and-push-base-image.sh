#!/bin/bash

IMAGE=node
TAG=14.16.1-stretch-slim-v1

docker pull docker.pkg.github.com/porscheui/porsche-design-system/$IMAGE:latest

docker build -t docker.pkg.github.com/porscheui/porsche-design-system/$IMAGE:$TAG .

docker push docker.pkg.github.com/porscheui/porsche-design-system/$IMAGE:$TAG
