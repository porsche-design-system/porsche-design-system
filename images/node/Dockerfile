FROM node:11.12.0-stretch@sha256:53caf7b35b949f365fdec7b599e495f125d6e3a54104caf05594a82e01343146

ENV PATH="/opt/porsche-ui-kit/bin:${PATH}"

RUN apt-get update && \
    apt-get install -yq gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 \
    libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 \
    libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 \
    libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 \
    ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils && \
    rm -rf /var/lib/apt/lists/*

COPY ./entrypoint.sh "/opt/porsche-ui-kit/entrypoint.sh"
COPY ./bin "/opt/porsche-ui-kit/bin"
COPY ./shared "/opt/porsche-ui-kit/shared"

ENTRYPOINT [ "/opt/porsche-ui-kit/entrypoint.sh" ]
