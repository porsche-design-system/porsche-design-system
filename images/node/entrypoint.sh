#!/usr/bin/env bash

set -o errexit
set -o pipefail

exec "${@}"
