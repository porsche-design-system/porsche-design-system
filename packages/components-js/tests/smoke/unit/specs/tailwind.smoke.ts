import fs from 'node:fs';
import { createRequire } from 'node:module';
import path from 'node:path';
import { describe, expect, test } from 'vitest';

const nodeRequire = createRequire(import.meta.url);

describe('ag-grid package content', () => {
  test('should match content', () => {
    const pkgPath = nodeRequire.resolve('@porsche-design-system/components-js');
    const themeCssPath = path.join(pkgPath, '../../tailwind/theme.css');
    const content = fs.readFileSync(themeCssPath, 'utf8');

    expect(content).toMatchSnapshot();
  });
});
