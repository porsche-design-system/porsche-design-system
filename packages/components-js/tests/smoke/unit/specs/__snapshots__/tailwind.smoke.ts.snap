// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ag-grid package content > should match content 1`] = `
"@custom-variant dark (&:where(.dark, .dark *));
@theme {
  --color-primary: #010205;
  --color-background-base: #fff;
  --color-background-surface: #eeeff2;
  --color-background-shading: rgba(1, 2, 5, 0.67);
  --color-background-frosted: hsla(240, 4%, 85%, 0.35);
  --color-contrast-low: #d8d8db;
  --color-contrast-medium: #6b6d70;
  --color-contrast-high: #535457;
  --color-notification-success: #197e10;
  --color-notification-success-soft: #e4ffec;
  --color-notification-warning: #f3be00;
  --color-notification-warning-soft: #fff4d2;
  --color-notification-error: #cc1922;
  --color-notification-error-soft: #ffe2e4;
  --color-notification-info: #2762ec;
  --color-notification-info-soft: #d3e1ff;
  --color-state-hover: rgba(148, 149, 152, 0.18);
  --color-state-active: rgba(148, 149, 152, 0.2);
  --color-state-focus: #1a44ea;
  --color-state-disabled: #949598;
  --font-porsche-next: Porsche Next, Arial Narrow, Arial, Heiti SC, SimHei, sans-serif;
  --font-weight-normal: 400;
  --font-weight-semibold: 600;
  --font-weight-bold: 700;
  --leading-normal: calc(6px + 2.125ex);
  --text-2xs: 0.75rem;
  --text-2xs--line-height: calc(6px + 2.125ex);
  --text-xs: clamp(0.81rem, 0.23vw + 0.77rem, 0.88rem);
  --text-xs--line-height: calc(6px + 2.125ex);
  --text-base: 1rem;
  --text-base--line-height: calc(6px + 2.125ex);
  --text-sm: 1rem;
  --text-sm--line-height: calc(6px + 2.125ex);
  --text-md: clamp(1.13rem, 0.21vw + 1.08rem, 1.33rem);
  --text-md--line-height: calc(6px + 2.125ex);
  --text-lg: clamp(1.27rem, 0.51vw + 1.16rem, 1.78rem);
  --text-lg--line-height: calc(6px + 2.125ex);
  --text-xl: clamp(1.42rem, 0.94vw + 1.23rem, 2.37rem);
  --text-xl--line-height: calc(6px + 2.125ex);
  --text-2xl: clamp(1.6rem, 1.56vw + 1.29rem, 3.16rem);
  --text-2xl--line-height: calc(6px + 2.125ex);
  --breakpoint-xs: 480px;
  --breakpoint-sm: 760px;
  --breakpoint-md: 1000px;
  --breakpoint-lg: 1300px;
  --breakpoint-xl: 1760px;
  --breakpoint-2xl: 1920px;
  --spacing: 0.25rem;
  --spacing-fluid-xs: clamp(4px, 0.25vw + 3px, 8px);
  --spacing-fluid-sm: clamp(8px, 0.5vw + 6px, 16px);
  --spacing-fluid-md: clamp(16px, 1.25vw + 12px, 36px);
  --spacing-fluid-lg: clamp(32px, 2.75vw + 23px, 76px);
  --spacing-fluid-xl: clamp(48px, 3vw + 38px, 96px);
  --spacing-fluid-2xl: clamp(80px, 7.5vw + 56px, 200px);
  --spacing-static-xs: 4px;
  --spacing-static-sm: 8px;
  --spacing-static-md: 16px;
  --spacing-static-lg: 32px;
  --spacing-static-xl: 48px;
  --spacing-static-2xl: 80px;
  --radius-sm: 4px;
  --radius-md: 8px;
  --radius-lg: 12px;
  --ease-base: cubic-bezier(0.25, 0.1, 0.25, 1);
  --ease-in: cubic-bezier(0, 0, 0.2, 1);
  --ease-out: cubic-bezier(0.4, 0, 0.5, 1);
  --default-border-width: 2px;
  --default-outline-width: 2px;
  --default-transition-timing-function: cubic-bezier(0.25, 0.1, 0.25, 1);
  --default-transition-duration: 0.25s;
  /** Custom none namespace variables (https://tailwindcss.com/docs/theme#theme-variable-namespaces) */
  --border-width-thin: 1px;
  --border-width-base: 2px;
  --duration-short: 0.25s;
  --duration-moderate: 0.4s;
  --duration-long: 0.6s;
  --duration-very-long: 1.2s;
  --drop-shadow-high: 0px 8px 40px rgba(0, 0, 0, 0.16);
  --drop-shadow-medium: 0px 4px 16px rgba(0, 0, 0, 0.16);
  --drop-shadow-low: 0px 3px 8px rgba(0, 0, 0, 0.16);
  --frosted-glass: 32px;
  --gradient-to-bottom: to bottom,
  hsla(0, 0%, 0%, 0.8) 0%,
  hsla(0, 0%, 0%, 0.8) 8.1%,
  hsla(0, 0%, 0%, 0.8) 15.5%,
  hsla(0, 0%, 0%, 0.8) 22.5%,
  hsla(0, 0%, 0%, 0.78) 29%,
  hsla(0, 0%, 0%, 0.73) 35.3%,
  hsla(0, 0%, 0%, 0.67) 41.2%,
  hsla(0, 0%, 0%, 0.6) 47.1%,
  hsla(0, 0%, 0%, 0.52) 52.9%,
  hsla(0, 0%, 0%, 0.44) 58.8%,
  hsla(0, 0%, 0%, 0.33) 64.7%,
  hsla(0, 0%, 0%, 0.22) 71%,
  hsla(0, 0%, 0%, 0.12) 77.5%,
  hsla(0, 0%, 0%, 0.05) 84.5%,
  hsla(0, 0%, 0%, 0.011) 91.9%,
  hsla(0, 0%, 0%, 0) 100%;
  --gradient-to-left: to left,
  hsla(0, 0%, 0%, 0.8) 0%,
  hsla(0, 0%, 0%, 0.8) 8.1%,
  hsla(0, 0%, 0%, 0.8) 15.5%,
  hsla(0, 0%, 0%, 0.8) 22.5%,
  hsla(0, 0%, 0%, 0.78) 29%,
  hsla(0, 0%, 0%, 0.73) 35.3%,
  hsla(0, 0%, 0%, 0.67) 41.2%,
  hsla(0, 0%, 0%, 0.6) 47.1%,
  hsla(0, 0%, 0%, 0.52) 52.9%,
  hsla(0, 0%, 0%, 0.44) 58.8%,
  hsla(0, 0%, 0%, 0.33) 64.7%,
  hsla(0, 0%, 0%, 0.22) 71%,
  hsla(0, 0%, 0%, 0.12) 77.5%,
  hsla(0, 0%, 0%, 0.05) 84.5%,
  hsla(0, 0%, 0%, 0.011) 91.9%,
  hsla(0, 0%, 0%, 0) 100%;
  --gradient-to-right: to right,
  hsla(0, 0%, 0%, 0.8) 0%,
  hsla(0, 0%, 0%, 0.8) 8.1%,
  hsla(0, 0%, 0%, 0.8) 15.5%,
  hsla(0, 0%, 0%, 0.8) 22.5%,
  hsla(0, 0%, 0%, 0.78) 29%,
  hsla(0, 0%, 0%, 0.73) 35.3%,
  hsla(0, 0%, 0%, 0.67) 41.2%,
  hsla(0, 0%, 0%, 0.6) 47.1%,
  hsla(0, 0%, 0%, 0.52) 52.9%,
  hsla(0, 0%, 0%, 0.44) 58.8%,
  hsla(0, 0%, 0%, 0.33) 64.7%,
  hsla(0, 0%, 0%, 0.22) 71%,
  hsla(0, 0%, 0%, 0.12) 77.5%,
  hsla(0, 0%, 0%, 0.05) 84.5%,
  hsla(0, 0%, 0%, 0.011) 91.9%,
  hsla(0, 0%, 0%, 0) 100%;
  --gradient-to-top: to top,
  hsla(0, 0%, 0%, 0.8) 0%,
  hsla(0, 0%, 0%, 0.8) 8.1%,
  hsla(0, 0%, 0%, 0.8) 15.5%,
  hsla(0, 0%, 0%, 0.8) 22.5%,
  hsla(0, 0%, 0%, 0.78) 29%,
  hsla(0, 0%, 0%, 0.73) 35.3%,
  hsla(0, 0%, 0%, 0.67) 41.2%,
  hsla(0, 0%, 0%, 0.6) 47.1%,
  hsla(0, 0%, 0%, 0.52) 52.9%,
  hsla(0, 0%, 0%, 0.44) 58.8%,
  hsla(0, 0%, 0%, 0.33) 64.7%,
  hsla(0, 0%, 0%, 0.22) 71%,
  hsla(0, 0%, 0%, 0.12) 77.5%,
  hsla(0, 0%, 0%, 0.05) 84.5%,
  hsla(0, 0%, 0%, 0.011) 91.9%,
  hsla(0, 0%, 0%, 0) 100%;
  --pds-grid-narrow-column-start: narrow-start;
  --pds-grid-narrow-column-end: narrow-end;
  --pds-grid-narrow-span-one-half: var(--pds-grid-narrow-span-one-half);
  --pds-grid-narrow-offset: max(22px, 10.625vw - 12px);
  --pds-grid-narrow-offset-sm: calc(5vw - 16px + (clamp(16px, 1.25vw + 12px, 36px) + (100vw - (5vw - 16px) * 2 - clamp(16px, 1.25vw + 12px, 36px) * 15) / 16) * 4);
  --pds-grid-narrow-offset-2xl: calc(max(0px, 50vw - 1280px) + min(50vw - 880px, 400px) + (clamp(16px, 1.25vw + 12px, 36px) + (min(100vw, 2560px) - min(50vw - 880px, 400px) * 2 - clamp(16px, 1.25vw + 12px, 36px) * 15) / 16) * 4);
  --pds-grid-basic-column-start: basic-start;
  --pds-grid-basic-column-end: basic-end;
  --pds-grid-basic-span-one-half: var(--pds-grid-basic-span-one-half);
  --pds-grid-basic-span-one-third: var(--pds-grid-basic-span-one-third);
  --pds-grid-basic-span-two-thirds: var(--pds-grid-basic-span-two-thirds);
  --pds-grid-basic-offset: max(22px, 10.625vw - 12px);
  --pds-grid-basic-offset-sm: calc(5vw - 16px + (clamp(16px, 1.25vw + 12px, 36px) + (100vw - (5vw - 16px) * 2 - clamp(16px, 1.25vw + 12px, 36px) * 15) / 16) * 2);
  --pds-grid-basic-offset-2xl: calc(max(0px, 50vw - 1280px) + min(50vw - 880px, 400px) + (clamp(16px, 1.25vw + 12px, 36px) + (min(100vw, 2560px) - min(50vw - 880px, 400px) * 2 - clamp(16px, 1.25vw + 12px, 36px) * 15) / 16) * 2);
  --pds-grid-extended-column-start: extended-start;
  --pds-grid-extended-column-end: extended-end;
  --pds-grid-extended-span-one-half: var(--pds-grid-extended-span-one-half);
  --pds-grid-extended-offset: max(22px, 10.625vw - 12px);
  --pds-grid-extended-offset-sm: calc(5vw - 16px + (clamp(16px, 1.25vw + 12px, 36px) + (100vw - (5vw - 16px) * 2 - clamp(16px, 1.25vw + 12px, 36px) * 15) / 16) * 1);
  --pds-grid-extended-offset-2xl: calc(max(0px, 50vw - 1280px) + min(50vw - 880px, 400px) + (clamp(16px, 1.25vw + 12px, 36px) + (min(100vw, 2560px) - min(50vw - 880px, 400px) * 2 - clamp(16px, 1.25vw + 12px, 36px) * 15) / 16) * 1);
  --pds-grid-wide-column-start: wide-start;
  --pds-grid-wide-column-end: wide-end;
  --pds-grid-wide-offset: max(22px, 10.625vw - 12px);
  --pds-grid-wide-offset-sm: calc(5vw - 16px);
  --pds-grid-wide-offset-2xl: calc(max(0px, 50vw - 1280px) + min(50vw - 880px, 400px));
  --pds-grid-full-column-start: full-start;
  --pds-grid-full-column-end: full-end;
  --pds-grid-full-offset: max(0px, 50vw - 1280px);
  @keyframes skeletonAnimation {
    from {
      background-position-x: 100%;
    }
    to {
      background-position-x: -100%;
    }
  }
}
@layer theme {
  .light {
    --color-primary: #010205;
    --color-background-base: #fff;
    --color-background-surface: #eeeff2;
    --color-background-shading: rgba(1, 2, 5, 0.67);
    --color-background-frosted: hsla(240, 4%, 85%, 0.35);
    --color-contrast-low: #d8d8db;
    --color-contrast-medium: #6b6d70;
    --color-contrast-high: #535457;
    --color-notification-success: #197e10;
    --color-notification-success-soft: #e4ffec;
    --color-notification-warning: #f3be00;
    --color-notification-warning-soft: #fff4d2;
    --color-notification-error: #cc1922;
    --color-notification-error-soft: #ffe2e4;
    --color-notification-info: #2762ec;
    --color-notification-info-soft: #d3e1ff;
    --color-state-hover: rgba(148, 149, 152, 0.18);
    --color-state-active: rgba(148, 149, 152, 0.2);
    --color-state-focus: #1a44ea;
    --color-state-disabled: #949598;
  }
  .dark {
    --color-primary: #fbfcff;
    --color-background-base: #0e0e12;
    --color-background-surface: #212225;
    --color-background-shading: rgba(38, 38, 41, 0.67);
    --color-background-frosted: hsla(240, 3%, 26%, 0.35);
    --color-contrast-low: #404044;
    --color-contrast-medium: #88898c;
    --color-contrast-high: #afb0b3;
    --color-notification-success: #09d087;
    --color-notification-success-soft: #003320;
    --color-notification-warning: #f7cb47;
    --color-notification-warning-soft: #362b0a;
    --color-notification-error: #fc4040;
    --color-notification-error-soft: #3a0f0f;
    --color-notification-info: #178bff;
    --color-notification-info-soft: #04294e;
    --color-state-hover: rgba(148, 149, 152, 0.18);
    --color-state-active: rgba(126, 127, 130, 0.2);
    --color-state-focus: #1a44ea;
    --color-state-disabled: #7e7f82;
  }
  @media (prefers-color-scheme: dark) {
    .auto {
      --color-primary: #fbfcff;
      --color-background-base: #0e0e12;
      --color-background-surface: #212225;
      --color-background-shading: rgba(38, 38, 41, 0.67);
      --color-background-frosted: hsla(240, 3%, 26%, 0.35);
      --color-contrast-low: #404044;
      --color-contrast-medium: #88898c;
      --color-contrast-high: #afb0b3;
      --color-notification-success: #09d087;
      --color-notification-success-soft: #003320;
      --color-notification-warning: #f7cb47;
      --color-notification-warning-soft: #362b0a;
      --color-notification-error: #fc4040;
      --color-notification-error-soft: #3a0f0f;
      --color-notification-info: #178bff;
      --color-notification-info-soft: #04294e;
      --color-state-hover: rgba(148, 149, 152, 0.18);
      --color-state-active: rgba(126, 127, 130, 0.2);
      --color-state-focus: #1a44ea;
      --color-state-disabled: #7e7f82;
    }
  }
}
@layer components {
  .pds-grid {
    margin: 0 var(--pds-internal-grid-margin, 0);
    display: grid;
    padding: 0 calc(50% - var(--pds-internal-grid-margin, 0px) - 1280px);
    grid-gap: clamp(16px, 1.25vw + 12px, 36px);
    max-width: var(--pds-internal-grid-width-max, 2560px);
    min-width: var(--pds-internal-grid-width-min, 320px);
    box-sizing: content-box;
    grid-template-columns: [full-start] minmax(0, var(--pds-internal-grid-outer-column, calc(var(--pds-internal-grid-safe-zone) - clamp(16px, 1.25vw + 12px, 36px)))) [wide-start extended-start basic-start narrow-start] repeat(6, minmax(0, 1fr)) [narrow-end basic-end extended-end wide-end] minmax(0, var(--pds-internal-grid-outer-column, calc(var(--pds-internal-grid-safe-zone) - clamp(16px, 1.25vw + 12px, 36px)))) [full-end];
    --pds-internal-grid-safe-zone: max(22px, 10.625vw - 12px);
    --pds-grid-basic-span-one-half: span 3;
    --pds-grid-basic-span-one-third: span 2;
    --pds-grid-narrow-span-one-half: span 3;
    --pds-grid-basic-span-two-thirds: span 4;
    --pds-grid-extended-span-one-half: span 3;
  }
  @media (min-width: 760px) {
    .pds-grid {
      grid-template-columns: [full-start] minmax(0, var(--pds-internal-grid-outer-column, calc(var(--pds-internal-grid-safe-zone) - clamp(16px, 1.25vw + 12px, 36px)))) [wide-start] minmax(0, 1fr) [extended-start] minmax(0, 1fr) [basic-start] repeat(2, minmax(0, 1fr)) [narrow-start] repeat(8, minmax(0, 1fr)) [narrow-end] repeat(2, minmax(0, 1fr)) [basic-end] minmax(0, 1fr) [extended-end] minmax(0, 1fr) [wide-end] minmax(0, var(--pds-internal-grid-outer-column, calc(var(--pds-internal-grid-safe-zone) - clamp(16px, 1.25vw + 12px, 36px)))) [full-end];
      --pds-internal-grid-safe-zone: calc(5vw - 16px);
      --pds-grid-basic-span-one-half: span 6;
      --pds-grid-basic-span-one-third: span 4;
      --pds-grid-narrow-span-one-half: span 4;
      --pds-grid-basic-span-two-thirds: span 8;
      --pds-grid-extended-span-one-half: span 7;
    }
  }
  @media (min-width: 1920px) {
    .pds-grid {
      --pds-internal-grid-safe-zone: min(50vw - 880px, 400px);
    }
  }
}
@utility pds-text-2xs {
  @apply font-porsche-next not-italic font-normal text-2xs text-primary;
}
@utility pds-text-xs {
  @apply font-porsche-next not-italic font-normal text-xs text-primary;
}
@utility pds-text-sm {
  @apply font-porsche-next not-italic font-normal text-sm text-primary;
}
@utility pds-text-md {
  @apply font-porsche-next not-italic font-normal text-md text-primary;
}
@utility pds-text-lg {
  @apply font-porsche-next not-italic font-normal text-lg text-primary;
}
@utility pds-text-xl {
  @apply font-porsche-next not-italic font-normal text-xl text-primary;
}
@utility pds-text-2xl {
  @apply font-porsche-next not-italic font-normal text-2xl text-primary;
}
@utility pds-heading-sm {
  @apply font-porsche-next not-italic font-semibold text-sm text-primary;
}
@utility pds-heading-md {
  @apply font-porsche-next not-italic font-semibold text-md text-primary;
}
@utility pds-heading-lg {
  @apply font-porsche-next not-italic font-semibold text-lg text-primary;
}
@utility pds-heading-xl {
  @apply font-porsche-next not-italic font-semibold text-xl text-primary;
}
@utility pds-heading-2xl {
  @apply font-porsche-next not-italic font-semibold text-2xl text-primary;
}
@utility pds-display-sm {
  @apply font-porsche-next not-italic font-normal leading-normal text-primary;
  font-size: clamp(1.8rem, 2.41vw + 1.32rem, 4.21rem);
}
@utility pds-display-md {
  @apply font-porsche-next not-italic font-normal leading-normal text-primary;
  font-size: clamp(2.03rem, 3.58vw + 1.31rem, 5.61rem);
}
@utility pds-display-lg {
  @apply font-porsche-next not-italic font-normal leading-normal text-primary;
  font-size: clamp(2.28rem, 5.2vw + 1.24rem, 7.48rem);
}
@utility pds-skeleton {
  background: #eeeff2 linear-gradient(to right, transparent 0%, #f7f7f7 20%, transparent 50%) 0 0/200% 100%;
  display: block;
  border-radius: 4px;
  animation: skeletonAnimation 0.6s cubic-bezier(0.25, 0.1, 0.25, 1) infinite;
  @keyframes skeletonAnimation {
    from {
      background-position-x: 100%;
    }
    to {
      background-position-x: -100%;
    }
  }
  .dark {
    background: #212225 linear-gradient(to right, transparent 0%, #1a1b1e 20%, transparent 50%) 0 0/200% 100%;
    display: block;
    border-radius: 4px;
    animation: skeletonAnimation 0.6s cubic-bezier(0.25, 0.1, 0.25, 1) infinite;
  }
  @keyframes skeletonAnimation {
    from {
      background-position-x: 100%;
    }
    to {
      background-position-x: -100%;
    }
  }
  @media (prefers-color-scheme: dark) {
    .auto {
      background: #212225 linear-gradient(to right, transparent 0%, #1a1b1e 20%, transparent 50%) 0 0/200% 100%;
      display: block;
      border-radius: 4px;
      animation: skeletonAnimation 0.6s cubic-bezier(0.25, 0.1, 0.25, 1) infinite;
    }
    @keyframes skeletonAnimation {
      from {
        background-position-x: 100%;
      }
      to {
        background-position-x: -100%;
      }
    }
  }
}
"
`;
